### PR TITLE
Check identical solutions for exercise

### PR DESCRIPTION
**Pull request by gal432**

### CODE REVIEW

File: lms/lmsweb/views.py [hunk 1]
```diff
@@ -260,7 +261,7 @@ def comment():
         return fail(401, "You aren't allowed to watch this page.")
 
     if act == 'fetch':
-        return jsonify(Comment.by_solution(solution_id))
+        return jsonify(Comment.solution_dicts(solution_id))
```
Comment (yammesicka) [inline lms/lmsweb/views.py:264]: `get_solutions()` (Use verb, don't use Hungarian notation)

File: lms/lmstests/public/identical_tests/services.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                from_solution=self.solution,
+                to_solution=solution,
+            )
+
+    @staticmethod
+    def _clone_solution_comments(
+            from_solution: models.Solution,
+            to_solution: models.Solution,
+    ) -> None:
+        for comment in models.Comment.by_solution(
```
Comment (yammesicka) [inline lms/lmstests/public/identical_tests/services.py:61]: Please reformat this lines to be more readable

File: lms/lmstests/public/identical_tests/services.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                'fork the comments and solve',
+                self.solution.id, solution.id,
+            )
+            self._clone_solution_comments(
+                from_solution=solution,
+                to_solution=self.solution,
+            )
+            break
+
+    def check_if_can_solve_other_solutions(self):
```
Comment (yammesicka) [inline lms/lmstests/public/identical_tests/services.py:43]: _is_can_autocheck_other_solution

Comment (yammesicka) [reply lms/lmstests/public/identical_tests/services.py:43]: (Broken English, but much more coherent)

File: lms/lmstests/public/identical_tests/services.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        self._logger = logger
+
+    def initialize(self):
+        self._solution = models.Solution.get_by_id(self._solution_id)
+
+    @property
+    def solution(self) -> models.Solution:
+        return self._solution
+
+    def check_identical(self):
```
Comment (yammesicka) [inline lms/lmstests/public/identical_tests/services.py:24]: clone_comments_for_identical

File: lms/lmstests/public/identical_tests/services.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        for solution in models.Solution.select().join(
```
Comment (yammesicka) [inline lms/lmstests/public/identical_tests/services.py:25]: It's better to split the check to another method we call `check_identical`

Comment (gal432) [reply lms/lmstests/public/identical_tests/services.py:25]: Not sure I understand

File: lms/lmstests/public/identical_tests/services.py (hunk 3 above)
Comment (gal432) [reply lms/lmstests/public/identical_tests/services.py]: 1. Public method
2. I tried something else, WDYT?

File: lms/lmsweb/views.py [hunk 6]
```diff
@@ -390,6 +392,7 @@ def done_checking(exercise_id, solution_id):
         is_checked=True, checker=current_user.id,
     ).where(requested_solution)
     is_updated = changes.execute() == 1
+    identical_tests_tasks.solve_solution_with_identical_code.apply_async(args=(solution_id,))
```
Comment (yammesicka) [inline lms/lmsweb/views.py:395]: \>79

File: lms/lmsweb/views.py [hunk 7]
```diff
@@ -347,6 +348,7 @@ def upload():
             json_data_str=code,
         )
         flake8_tasks.run_flake8_on_solution.apply_async(args=(solution.id,))
+        identical_tests_tasks.solve_solution_with_identical_code.apply_async(args=(solution.id,))
```
Comment (yammesicka) [inline lms/lmsweb/views.py:351]: \>79

File: lms/lmstests/public/identical_tests/services.py (hunk 5 above)
Comment (yammesicka) [reply lms/lmstests/public/identical_tests/services.py:25]: The function does 2 things (SRP violation): checks if something is identical and also clones the comment

**Pull request merged**